### ISSUE TITLE
Galaxy S4 mini

### ISSUE THREAD

**Issue by AndroidUserwithMiuiu** [Seva-coder/Finder#33]
Hello,

the Samsung S4 Galaxy S4 Mini with Android 4.4.2 wont be enable GPS automatically. I have execute the ADB-Command  "adb shell pm grant ru.seva.finder android.permission.WRITE_SECURE_SETTINGS" and after this i have enable the Option. But it not works. If i enable GPS manual, it work. What can i do to enable GPS automatically with a SMS-Request?

**Comment by Seva-coder** [comment]
Hello, yes this feature sometimes has problems on some devices (but on certain it works normal). I have checked it on android 4.4 emulator... and it really not works! We have discussing it [here](https://github.com/Seva-coder/Finder/issues/7) and contributor (@domkrm) is there too. Moreover, such problem appeared on S7 too.
In logcat now it looks like app ghas not enough permissions, despite the fact that they are granted.
I will try to solve it

**Comment by domkrm** [comment]
I also have checked it on an Android 4.4 emulator and it seems for older Android versions is an additional permission needed `android.permission.WRITE_SETTINGS`.
Tomorrow I will check if this has any impacts on newer Android versions. Then I will make a pull request to fix this issue.

**Comment by Seva-coder** [comment]
it would be great if so!
@domkrm, thanks for support)

**Comment by AndroidUserwithMiuiu** [comment]
I'm curious. Thanks for the work

Am 03.03.2020 20:49 schrieb Sevastyanov Nikita:

**Comment by Seva-coder** [comment]
added this new right to the manifest and when requesting rights - GPS now turns on automatically (19 API). But there are problems with sending SMS twice (only when GPS is automatically turned on). May be it is because of emulator, but more likely due to my bad code ;)
I will try to fix

**Comment by Seva-coder** [comment]
I solved this problem - new permission "android.permission.WRITE_SETTINGS" added only to manifest, so it really grants only on API<23 (on new versions it should be granted via special window in system manually by user).
Also I got a strange bug in android 4.4 emulator - after calling method `removeUpdates` locations still gets to the listener.. Emulator send second location data with pause 10-20 ms since first, and it looks like `removeUpdates` is too slow, and it leads to second adding coordinates to SMS. I created boolean to skip such situation, and now it work reliable (I tested on 4.4, 6 and 8.0)
create release to F-Droid?

**Comment by Seva-coder** [comment]
Offtopic - but second build cycle in logs of F-droid ends with `"could not bring up vm 'builder_default'"`
Other apps also have such message. I i looked [here](https://f-droid.org/wiki/index.php?title=Special:RecentChanges&days=7&from=&hidebots=0&hideanons=1&hideliu=1&limit=500)

**Comment by AndroidUserwithMiuiu** [comment]
What do you mean?


Am 9. März 2020 16:41:34 schrieb Sevastyanov Nikita <notifications@github.com>:

**Comment by Seva-coder** [comment]
I meant that F-Droid build new versions very slow due to its internal errors..

**Comment by AndroidUserwithMiuiu** [comment]
You mean, in fdroid comes a new version with the fix?


Am 10. März 2020 18:33:51 schrieb Sevastyanov Nikita <notifications@github.com>:

**Comment by Seva-coder** [comment]
Of course, after testing this changes on some android versions, I marked it as a release for android. (but it still not updated there)
Yesterday I had asked them about problem with autoupdates of apps (in their [gitlab](https://gitlab.com/fdroid/fdroidserver/-/issues/752)), they know about it. I hope they fix this and apps began build normally

**Comment by AndroidUserwithMiuiu** [comment]
Ans when is the new version available in the fdroid store?


Am 11. März 2020 18:29:42 schrieb Sevastyanov Nikita <notifications@github.com>:

**Comment by Seva-coder** [comment]
I looked now build log - build [successfull!](https://f-droid.org/wiki/page/ru.seva.finder/lastbuild_13) Now wait for deploy, and I think may be tomorrow app will be available on their official site (in f-droid's app it can be available earlier)
Latest version now is v2.2
Also, now I read about how to add Travis CI - this service can build opensource apps after every commit. It may be faster in such cases

**Comment by Seva-coder** [comment]
f-droid updated it!
If you will test new version - please, give feedback about how it works/not works (GPS auto enable)

**Comment by AndroidUserwithMiuiu** [comment]
Ok. Thank you. And when its the update available in google play store?

Regards


Am 12. März 2020 19:46:46 schrieb Sevastyanov Nikita <notifications@github.com>:

**Comment by Seva-coder** [comment]
Unfortunately, google changed its rules for apps. Now (after spring 2019) apps with SMS_SEND permission must be an app for messaging, default app in system for sms.
Finder was published before this changes in Play store, but then has been deleted by google.. Changes described here - https://play.google.com/about/privacy-security-deception/permissions/  and https://support.google.com/googleplay/android-developer/answer/9047303#intended
It seems that we can try use exception - "Physical safety / emergency alerts to send SMS", but next saying that "Family or device locator" won't be permitted...
This event gone unnoticed by users, only developers felt it on itself. Google even held a webinar, when it happened. But it wasn't nothing useful in their answers :)
I think, I can try to publish our app to amazon store, yandex, samsung and may be xiaomi with huawei. But F-droid is the best, because its users can better describe (or even fix!) bug witch they faced. When most of bugs fixed, it is time to expand.

**Comment by AndroidUserwithMiuiu** [comment]
ok. i understand. Then i downloaded over f-drod.
Thank you.

Regards

Am 13.03.2020 21:02 schrieb Sevastyanov Nikita:

**Comment by darhma** [comment]
I've a redmi note 7 with lineageos 17.1 and magisk without gapps or microg. Installed the app (version 2.3) from f-droid the GPS auto enable doesn't work. The responding message says that the GPS is disabled

**Comment by domkrm** [comment]
> I've a redmi note 7 with lineageos 17.1 and magisk without gapps or microg. Installed the app (version 2.3) from f-droid the GPS auto enable doesn't work. The responding message says that the GPS is disabled

Do you have followed the instructions in the README for auto activating GPS?

https://github.com/Seva-coder/Finder#activate-location-automatically

**Comment by darhma** [comment]
@domkrm 
Yes but I've used a root terminal to give the command and not adb. The command was 'pm grant ru.seva.finder android.permission.WRITE_SECURE_SETTINGS' and after that I can select the option 'turn location on automatically' (previously it cannot be selected)

**Comment by AndroidUserwithMiuiu** [comment]
Hello,

the handy is not rooted. I have set following command via adb:
'pm grant ru.seva.finder android.permission.WRITE_SECURE_SETTINGS'

After this i have enabled the gps setting under settings.

Thank you for your support.

Regards


Am 12. August 2020 20:32:39 schrieb Dario Martines <notifications@github.com>:

**Comment by darhma** [comment]
@AndroidUserwithMiuiu 
The GPS auto enabling option works for you?

**Comment by AndroidUserwithMiuiu** [comment]
The option is enabled. But it doesnt work for me. See first mail..

Regards


Am 12. August 2020 22:00:02 schrieb Dario Martines <notifications@github.com>:

**Comment by AndroidUserwithMiuiu** [comment]
Any ideas? I have no miuiu.

Regards


Am 13. August 2020 07:09:46 schrieb anonyms@posteo.dk:

**Comment by AndroidUserwithMiuiu** [comment]
No solution for this Situation??

Regards


Am 14. August 2020 14:00:16 schrieb anonyms@posteo.dk:
